runtime/jcl/common/criusupport.c

```diff
@@ -41,6 +103,200 @@ Java_com_ibm_jvm_CRIUSupport_isCRIUSupportEnabledImpl(JNIEnv *env, jclass unused
 		}
 #endif /* defined(LINUX) */
 	}
+	setupJNIFieldIDs(env);
 
 	return res;
 }
+
+static BOOLEAN
+getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeString, char **nativeStringBuf, jfieldID *resultType)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	char mutf8StringBuf[STRING_BUFFER_SIZE];
+	char *mutf8String = NULL;
+	UDATA mutf8StringBufSize = 0;
+	UDATA requiredConvertedStringSize = 0;
+	BOOLEAN res = TRUE;
+	PORT_ACCESS_FROM_VMC(currentThread);
+
+	memset(mutf8StringBuf, 0, sizeof(mutf8StringBuf));
+
+	mutf8String = vmFuncs->copyStringToUTF8WithMemAlloc(currentThread, javaString, J9_STR_NULL_TERMINATE_RESULT, "", 0, mutf8StringBuf, STRING_BUFFER_SIZE, &mutf8StringBufSize);
+	if (NULL == mutf8String) {
```

Comment:
This seems overly restrictive, it should allocate a buffer if STRING_BUFFER_SIZE isn't big enough.

Reply:
oh right, ok.